scripts/dts/gen_defines.py

```diff
@@ -59,9 +59,6 @@ def main():
             if not node.enabled:
                 out_comment("No node macros: node is disabled")
```

Comment:
So if we're moving to https://github.com/zephyrproject-rtos/zephyr/pull/25002, I think this should be removed too, right? We had discussed being able to read any property we have semantics for (at least 'reg' and 'interrupts') regardless of status.

Reply:
I also think things like `compat2enabled` in the EDT object may need changes, since we're changing the semantics of "enabled" in C to be `status == "okay"` instead of `status != "disabled"` and adding macros like DT_HAS_COMPAT_STATUS_OKAY in https://github.com/zephyrproject-rtos/zephyr/pull/25002. So it seems like we should be doing `compat2statusokay` or something instead.

In general I'm having trouble keeping all the pull requests together and I don't see enough Python changes to accompany all the C changes. What am I missing?

Reply:
I reworked how enabled was defined.  If we want to leave that alone we can create a new 'available' which would be similar to the linux concept https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/of/base.c#n581